X9.146/gen_dual_keysig_cert.c

```diff
@@ -0,0 +1,375 @@
+ *
+ * This file is part of wolfSSL.
+ *
+ * wolfSSL is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * wolfSSL is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1335, USA
+ */
+
+#include <stdio.h>
+#include <wolfssl/options.h>
+#include <wolfssl/wolfcrypt/settings.h>
+#include <wolfssl/wolfcrypt/ecc.h>
+#include <wolfssl/wolfcrypt/rsa.h>
+#include <wolfssl/wolfcrypt/asn_public.h>
+#include <wolfssl/wolfcrypt/asn.h>
+#include <wolfssl/wolfcrypt/error-crypt.h>
+#include <wolfssl/wolfcrypt/logging.h>
+
+#if defined(WOLFSSL_X9_146)
+
+#define LARGE_TEMP_SZ 4096
+
+#if defined(GEN_ROOT_CERT) && defined(GEN_SERVER_CERT)
+    #error "Please only generate a root OR server certificate."
+#endif
+
+#define SUBJECT_COUNTRY "US"
+#define SUBJECT_STATE "MT"
+#define SUBJECT_LOCALITY "YourCity"
+#define SUBJECT_ORG "YourOrgName"
+#define SUBJECT_UNIT "YourUnitName"
+#define SUBJECT_COMMONNAME "www.YourDomain.com"
+
+#ifdef GEN_ROOT_CERT
+#define SUBJECT_EMAIL "root@YourDomain.com"
+#else
+#define SUBJECT_EMAIL "server@YourDomain.com"
+#endif
+
+static int do_certgen(int argc, char** argv)
+{
+    int ret = 0;
+
+    char caKeyFile[] = "./ca-key.der";
+#ifdef GEN_ROOT_CERT
+    char newCertOutput[] = "./ca-cert.der";
+    char sapkiFile[] = "./alt-ca-pub-key.der";
+    char altPrivFile[] = "./alt-ca-key.der";
+#else
+    char caCert[] = "./ca-cert.der";
+    char newCertOutput[] = "./server-cert.der";
+    char sapkiFile[] = "./alt-server-pub-key.der";
+    char altPrivFile[] = "./alt-server-key.der";
+    char serverKeyFile[] = "./server-key.der";
+#endif
+    FILE* file;
+    Cert newCert;
+    DecodedCert preTBS;
+
+#ifndef GEN_ROOT_CERT
+    byte caCertBuf[LARGE_TEMP_SZ];
+    int caCertSz = LARGE_TEMP_SZ;
+    byte serverKeyBuf[LARGE_TEMP_SZ];
+    int serverKeySz = LARGE_TEMP_SZ;
+#endif /* !GEN_ROOT_CERT */
+
+    byte caKeyBuf[LARGE_TEMP_SZ];
+    int caKeySz = LARGE_TEMP_SZ;
+    byte sapkiBuf[LARGE_TEMP_SZ];
+    int sapkiSz = LARGE_TEMP_SZ;
+    byte altPrivBuf[LARGE_TEMP_SZ];
+    int altPrivSz = LARGE_TEMP_SZ;
+    byte altSigAlgBuf[LARGE_TEMP_SZ];
+    int altSigAlgSz = LARGE_TEMP_SZ;
+    byte scratchBuf[LARGE_TEMP_SZ];
+    int scratchSz = LARGE_TEMP_SZ;
+    byte preTbsBuf[LARGE_TEMP_SZ];
+    int preTbsSz = LARGE_TEMP_SZ;
+    byte altSigValBuf[LARGE_TEMP_SZ];
+    int altSigValSz = LARGE_TEMP_SZ;
+    byte outBuf[LARGE_TEMP_SZ];
+    int outSz = LARGE_TEMP_SZ;

```

Comment:
That's a lot of stack usage here...do all of these need to be 4k? Is ok for example though....

Reply:
yeah, this simplifies the example.  Lets keep it like this please.